external/common/lapack/CMakeLists.txt

```diff
@@ -20,26 +19,42 @@ set_property(TARGET blas PROPERTY INTERFACE_LINK_LIBRARIES ${BLAS_LIBRARIES})
 set_property(TARGET lapk PROPERTY INTERFACE_LINK_LIBRARIES ${LAPACK_LIBRARIES})
 set_property(TARGET blas PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${BLAS_INCLUDE_DIRS})
 set_property(TARGET lapk PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${LAPACK_INCLUDE_DIRS})
-set_property(TARGET lapack PROPERTY INTERFACE_LINK_LIBRARIES lapk blas)
-get_property(_ill TARGET lapk PROPERTY INTERFACE_LINK_LIBRARIES)
+set_property(TARGET lapack PROPERTY INTERFACE_LINK_LIBRARIES lapk blas tgt::MathOpenMP)
 
 # <<  Examine library list for MKL-ness  >>
 foreach(_l IN LISTS LAPACK_LIBRARIES)
-    if(${_l} MATCHES "mkl")
-        set(_isMKL " MKL")
+    get_filename_component(_lname ${_l} NAME)
+    if(${_lname} MATCHES "mkl")
+        set(isMKL " MKL")
         target_compile_definitions(lapack INTERFACE -DUSING_LAPACK_MKL)
         break()
     endif()
 endforeach()
 
+# <<  Detect OpenMP and modify for BLAS/LAPACK  >>
+if(NOT TARGET tgt::MathOpenMP)
+    list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
+    find_package(MathOpenMP)
+endif()
+
+include(CMakePrintHelpers)
```

Comment:
Nice to always discover new CMake modules :smile_cat: Was this print meant for debugging only?

Reply:
In true CMake fashion, no one thought to put an `ALL` option so you don't have to list every property. Yes, ultimately this'll be debug printing, but in the review period, it's "check every time" printing.